core/src/main/java/com/rudderstack/android/sdk/core/util/Utils.java

```diff
@@ -94,8 +94,28 @@ public static String getDeviceId(Application application) {
         return null;
     }
 
+    @Nullable
+    private static String serializeObject(Object obj) {
+        String json;
+        if (obj instanceof String) {
+            json = (String) obj;
+        } else {
+            json = RudderGson.serialize(obj);
+        }
+        return json;
+    }
+
+    /**
+     * Convert an object to a map
+     * <p> Serialize the object to a json string and then convert it to a map </p>
+     * <p> If the object is a string, it is directly converted to a map </p>
+     * <p> If the object results in an invalid json string after serialization, an empty map is returned </p>
```

Comment:
Ideally, there is no need to document the implementation.

Resolve this comment with/without making the changes.